io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shelly25/IoShelly25Impl.java

```diff
@@ -114,54 +108,55 @@ public void handleEvent(Event event) {
 	 * Execute on Cycle Event "Before Process Image".
 	 */
 	private void eventBeforeProcessImage() {
-		Boolean relay1IsOn;
-		Boolean relay2IsOn;
-		try {
-			var json = this.shellyApi.getStatus();
-			var relays = JsonUtils.getAsJsonArray(json, "relays");
-			var relay1 = JsonUtils.getAsJsonObject(relays.get(0));
-			relay1IsOn = JsonUtils.getAsBoolean(relay1, "ison");
-			var relay2 = JsonUtils.getAsJsonObject(relays.get(1));
-			relay2IsOn = JsonUtils.getAsBoolean(relay2, "ison");
-
-			this._setSlaveCommunicationFailed(false);
-
-		} catch (OpenemsNamedException | IndexOutOfBoundsException e) {
-			relay1IsOn = null;
-			relay2IsOn = null;
-			this.logError(this.log, "Unable to read from Shelly API: " + e.getMessage());
-			this._setSlaveCommunicationFailed(true);
-		}
-		this._setRelay1(relay1IsOn);
-		this._setRelay2(relay2IsOn);
+		String url = this.baseUrl + "/status";
+		this.httpBridge.get(url).whenComplete((result, error) -> {
```

Comment:
for reading values this should not be called every cycle. The single use methods `get`, `post`, `delete`... are intended to be used rarely and only in certain situations.

In your case you want to read the value every cycle then you can use the `HttpBridge.subscribeJsonEveryCylce` method. Make sure to also remove unused subscribes to events after the change.

You can also have a look at the implemention in the [IoShelly3EmImpl.java](https://github.com/OpenEMS/openems/blob/develop/io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shelly3em/IoShelly3EmImpl.java)